### DISCUSSION TITLE
I ran systemctl restart netbox netbox-rq and now i keep getting error 502 on the UI. I am also having issues with netbox-scanner not inserting anything into netbox

### DISCUSSION THREAD

**Discussion by jburg379** [netbox-community/netbox#17019] [Help Wanted!]
### Deployment Type

Self-hosted

### NetBox Version

v4.0.7

### Python Version

3.12

### Steps to Reproduce

ran systemctl restart netbox netbox-rq
It asked me for the password 2 times and now the UI is showing error 502
I tried restarting the netbox server

### Expected Behavior

I just wanted to restart the netbox service since I changed the system time zone

### Observed Behavior

UI is showing error 502

**Comment by ZPrimed** [comment]
Output from `systemctl status netbox` and `systemctl status netbox-rq` would be helpful; my hunch is that one or both services didn't re-launch cleanly after whatever changes you made.

**Reply by jburg379** [reply]
netbox.service - NetBox WSGI Service
     Loaded: loaded (/etc/systemd/system/netbox.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Tue 2024-07-30 07:58:59 EDT; 23s ago
       Docs: https://docs.netbox.dev/
    Process: 33740 ExecStart=/opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox  --config /opt/netbox/gunicorn.py netbox.wsgi (code=exited, status=1/FAILURE)
   Main PID: 33740 (code=exited, status=1/FAILURE)
        CPU: 2.161s
...skipping...


systemctl status netbox-rq
● netbox-rq.service - NetBox Request Queue Worker
     Loaded: loaded (/etc/systemd/system/netbox-rq.service; enabled; preset: enabled)
     Active: activating (auto-restart) (Result: exit-code) since Tue 2024-07-30 07:59:49 EDT; 8s ago
       Docs: https://docs.netbox.dev/
    Process: 33987 ExecStart=/opt/netbox/venv/bin/python3 /opt/netbox/netbox/manage.py rqworker high default low (code=exited, status=1/FAILURE)
   Main PID: 33987 (code=exited, status=1/FAILURE)
        CPU: 407ms

**Reply by jburg379** [reply]
When I tried to restart it I forgot to do it using sudo so it asked me for the password. I just tried doing it again and it gave me the following message I think this might have happened the first time too

Warning! D-Bus connection terminated.
Failed to wait for response: Connection reset by peer

**Reply by ZPrimed** [reply]
You cut out all of the interesting lines that might explain why either of those services is in a **FAILURE** state. Both should be "active (running)" and not "activating" or showing anything that says "exited" or "FAILURE". In both cases it appears that they are being started, crashing, and then systemd is automatically re-launching them after the crash... but I suspect they are then crashing again.

You'll generally want to run any `systemctl` command with `sudo`, or else it will ask you to authenticate as an account that has root permissions.

**Reply by candlerb** [reply]
```
sudo journalctl -u netbox -n 200 --no-pager
sudo journalctl -u netbox-rq -n 200 --no-pager
```

**Reply by jburg379** [reply]
Took me a while but I was able to fix it. Not exactly sure how it happened but it looks like it started having some kind of permissions issue after restarting the netbox and netbox-rq services. I changed the owner of the /opt/netbox venv to netbox:netbox and I was able to re-install the dependencies and then it started working after that.

**Reply by markkuleinio** [reply]
I wouldn't leave the venv (or other NetBox files) to netbox:netbox because it means that the NetBox application (or user-created scripts) can then potentially modify itself, and it is thus a security issue.

By default the files are owned by root (because it was the root user that created the venv), and read-accessible (directories also executable) to all users.
